src/main/java/microsoft/exchange/webservices/data/ServiceRequestBase.java

```diff
@@ -621,13 +621,24 @@ protected HttpWebRequest validateAndEmitRequest() throws ServiceLocalException,
     this.validate();
 
     HttpWebRequest request = this.buildEwsHttpWebRequest();
+
     try {
-      return this.getEwsHttpWebResponse(request);
-    } catch (HttpErrorException e) {
-      processWebException(e, request);
+      try {
+        return this.getEwsHttpWebResponse(request);
+      } catch (HttpErrorException e) {
+        processWebException(e, request);
 
-      // Wrap exception if the above code block didn't throw
-      throw new ServiceRequestException(String.format(Strings.ServiceRequestFailed, e.getMessage()), e);
+        // Wrap exception if the above code block didn't throw
+        throw new ServiceRequestException(String.format(Strings.ServiceRequestFailed, e.getMessage()), e);
+      }
+    } catch (Exception e) {
```

Comment:
Yeah, it should. In all non-error cases it is properly closed, but in case of an error it is not returned, so cannot be closed by the caller. It's a bit weird, the whole error handling could probably use some refactoring in the future.